flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridge.java

```diff
@@ -354,7 +354,7 @@ private void failPendingRequests(Collection<ResourceRequirement> acquiredResourc
                     getSlotServiceStatus());
             cancelPendingRequests(
                     predicate,
-                    new NoResourceAvailableException(
+                    NoResourceAvailableException.withoutStackTrace(
```

Comment:
I think this can work. An alternative idea would be to introduce a sub class of `NoResourceAvailableException` that does not allow to set a cause.

Reply:
Actually, `withoutStackTrace()` makes it a bit more explicit compared to the sub class approach.

Reply:
I agree, having a dedicated method makes it more explicit.